repo/content/content_upgrade_lock.go

```diff
@@ -0,0 +1,160 @@
+package content
+
+import (
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+// UpgradeLock represents the intent to lock a kopia repository for upgrade
+// related maintenance activity. This signals a request for exclusive access to
+// the repository. The lock object is set on the Kopia repository format blob
+// 'kopia.repository' and must be respected by all clients accessing the
+// repository.
+type UpgradeLock struct {
+	OwnerID                string        `json:"ownerID,omitempty"`
+	CreationTime           time.Time     `json:"creationTime,omitempty"`
+	AdvanceNoticeDuration  time.Duration `json:"advanceNoticeDuration,omitempty"`
+	IODrainTimeout         time.Duration `json:"ioDrainTimeout,omitempty"`
```

Comment:
instead of time.Duration can this be a point in time (`IODrainTime time.Time`) ? Would it makes this simpler to reason about?

Reply:
I'd prefer that we always compute the `UpgradeTime()` from config parameters in the lock instead of pre-determining it. This makes it easier to alter parameters later (for example, postponing advance notice) and reason about why the `UpgradeTime()` is what it is at any given point which is good for debugging.